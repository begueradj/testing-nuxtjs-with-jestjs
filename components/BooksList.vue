<template>
  <v-container
    grid-list-md
    text-xs-center
    fill-height>
    <v-layout
      row
      wrap
      align-center>
      <v-flex
        xs>
        My favorite book is <h4 class="test-class"> {{ book }} </h4>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script>
export default {
  name: 'BooksList',
  data() {
    return {
      book: 'JavaScript'
    }
    
  }
}
</script>

<style scoped>
.test-class {
    color: red;
    background-color: yellow;
}
</style>
