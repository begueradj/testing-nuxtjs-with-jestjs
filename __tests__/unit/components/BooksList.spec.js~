import Vue from 'vue'
import { mount } from '@vue/test-utils'
import BooksList from '@/components/BooksList.vue'
import Vuetify from 'vuetify'
Vue.use(Vuetify)

describe("BooksList component", () => {
    it("renders correctly", () => {
        expect(true).toBe(true)
    })
})

describe("test 2", () => {
  it("test 2", () => {
    //const bookslist = new Vue(BooksList).$mount()
    // const bookslist = Vue.extend(BooksList)
    // const vm = new bookslist().$mount()
    // console.log(vm.book)
    const wrapper = mount(BooksList)
    const vm = wrapper.vm
    console.log(vm.book)
    // expect(vm.book).toContain('JavaScript')
  })
})

