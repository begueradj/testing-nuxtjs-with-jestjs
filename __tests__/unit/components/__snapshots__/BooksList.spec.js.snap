// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 2 test 2 1`] = `
<div
  class="container grid-list-md text-xs-center fill-height"
>
  <div
    class="layout row wrap align-center"
  >
    <div
      class="flex xs"
    >
      
      My favorite book is 
      <h4
        class="test-class"
      >
         Java 
      </h4>
    </div>
  </div>
</div>
`;
